src/Learning/KWUserInterface/CMakeLists.txt

```diff
@@ -1,7 +1,50 @@
 file(GLOB cppfiles ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+generate_gui_get_log_list(ddlogfiles)
+
 add_library(KWUserInterface STATIC "${cppfiles}")
+if(GENERATE_VIEWS)
+  target_sources(KWUserInterface PRIVATE "${ddlogfiles}" KWSelectionParameters.dd.log KWTrainParameters.dd.log)
```

Comment:
Why here we use both the log list function and the `if(GENERATE_VIEWS)` ?

Reply:
for the same reason: `KWSelectionParameters.dd.log `and `KWTrainParameters.dd.log` are not generated by `generate_gui_get_log_list` and we have to add them only if we generate views, otherwise, if those files are missing, the project configuration will fail

Reply:
Ok